tests/app/dao/test_services_dao.py

```diff
@@ -874,6 +874,93 @@ def test_fetch_stats_for_today_only_includes_today(notify_db_session):
+    with freeze_time('2021-03-28T00:01:00'):
+        # just after midnight yesterday in UTC -- included
+        create_notification(template=template, to_field='2', status='failed')
+    with freeze_time('2021-03-28T12:00:00'):
+        # we have entered BST at this point but had not for the previous two notifications --included
+        # collect stats for this timestamp
+        create_notification(template=template, to_field='3', status='created')
+        stats = dao_fetch_todays_stats_for_service(template.service_id)
+
+    stats = {row.status: row.count for row in stats}
+    assert 'delivered' not in stats
+    assert stats['failed'] == 1
+    assert stats['created'] == 1
+    assert not stats.get('permanent-failure')
+    assert not stats.get('temporary-failure')
+
+
+def test_fetch_stats_for_today_only_includes_today_during_bst(notify_db_session):
+    template = create_template(service=create_service())
+    with freeze_time('2021-03-28T22:59:59'):
+        # just before midnight BST -- not included
+        create_notification(template=template, to_field='1', status='permanent-failure')
+    with freeze_time('2021-03-28T23:00:01'):
+        # just after midnight BST -- included
+        create_notification(template=template, to_field='2', status='failed')
+    with freeze_time('2021-03-29T12:00:00'):
+        # well after midnight BST -- included
+        # collect stats for this timestamp
+        create_notification(template=template, to_field='3', status='created')
+        stats = dao_fetch_todays_stats_for_service(template.service_id)
+
+    stats = {row.status: row.count for row in stats}
+    assert 'delivered' not in stats
+    assert stats['failed'] == 1
+    assert stats['created'] == 1
+    assert not stats.get('permanent-failure')
+
+
+def test_fetch_stats_for_today_only_includes_today_when_clocks_fall_back(notify_db_session):
+    template = create_template(service=create_service())
+    with freeze_time('2021-10-30T22:59:59'):
+        # just before midnight BST -- not included
+        create_notification(template=template, to_field='1', status='permanent-failure')
+    with freeze_time('2021-10-31T23:00:01'):
+        # just after midnight BST -- included
+        create_notification(template=template, to_field='2', status='failed')
+    # clocks go back to UTC on 31 October at 2am
+    with freeze_time('2021-10-31T12:00:00'):
+        # well after midnight -- included
+        # collect stats for this timestamp
+        create_notification(template=template, to_field='3', status='created')
+        stats = dao_fetch_todays_stats_for_service(template.service_id)
+
+    stats = {row.status: row.count for row in stats}
+    assert 'delivered' not in stats
+    assert stats['failed'] == 1
+    assert stats['created'] == 1
+    assert not stats.get('permanent-failure')
+
+
+def test_fetch_stats_for_today_only_includes_during_utc(notify_db_session):
```

Comment:
Very optional: I think this test duplicates `test_fetch_stats_for_today_only_includes_today` but that this one has a better name. Could remove one of them/merge them together if you are fussed